ocfweb/api/stats.py

```diff
@@ -17,3 +23,58 @@ def get_staff_in_lab(request: HttpRequest) -> JsonResponse:
         staff_in_lab(),
         safe=False,
     )
+
+
+def get_printers_summary(request: HttpRequest) -> JsonResponse:
+    response = JsonResponse(
+        printers(),
+        safe=False,
+    )
+    return response
+
+
+def get_desktop_usage(request: HttpRequest) -> JsonResponse:
+    """
+    Copy desktop usage from Django API by grabbing class attributes that
+    can't be serialized to JSON
+    """
+
+    responseList = []
+    for profile in desktop_profiles():
+        minutes_idle = profile.minutes_idle
+        minutes_busy = profile.minutes_busy
+        responseList.append({
+            'hostname': profile.hostname,
+            'minutes_idle': int(minutes_idle),
+            'minutes_busy': int(minutes_busy),
+            'percent': int(100 * minutes_busy / max(1, (minutes_idle + minutes_busy))),
+        })
+
+    response = JsonResponse(
+        responseList,
+        safe=False,
+    )
+    return response
+
+
+def get_mirrors_showcase(request: HttpRequest) -> JsonResponse:
+    """ Return bandwidth for a few mirrors that we showcase on stats page
+    In human-readable form, sorted with biggest bandwidth first
+    """
+    mirrors_showcase = [['ubuntu', 0], ['debian', 0], ['archlinux', 0]]
```

Comment:
I think it probably has to be set mirrors because they are shown with that project's logo on ocfstatic, so we have to have the logos for mirrors we showcase

Reply:
Oh that makes sense!

Reply:
```suggestion
    mirrors_showcase: List[Tuple[str, int]] = [('ubuntu', 0), ('debian', 0), ('archlinux', 0)]
```

Mypy is confused below because it inferred the type of `mirrors_showcase` to be `List[List[object]]` since `object` is the only type both `str` and `int` inherit from. I think this should make things clearer and not confuse mypy.

Reply:
Also you might consider using a dictionary here instead of a list of tuples...

Reply:
I was thinking of using a union, such as `List[List[Union[str, int]]]` but I think tuples would fit better

Reply:
Yeah I would definitely use either a list of tuples or just a dictionary. The items inside the list are immutable and of a short length so tuples are a perfect fit.